packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -430,11 +444,14 @@ export class VueElement
       if (!hasOwn(this, key)) {
         // exposed properties are readonly
         Object.defineProperty(this, key, {
+          configurable: true, // should be configurable to allow deleting when disconnected
           // unwrap ref to be consistent with public instance behavior
           get: () => unref(exposed[key]),
         })
-      } else if (__DEV__) {
-        warn(`Exposed property "${key}" already exists on custom element.`)
+      } else {
+        delete exposed[key] // delete it from exposed in case of deleting wrong exposed key when disconnected
```

Comment:
This is to remove invalid exposed key. Let's say we have a element with prop `foo`, and we accidentally expose an object `foo`, it will be ignored as `foo` is already on this element. In disconnectedCallback, we'll delete exposed properties according to `exposed`, but deleting `foo` can cause an error as it's non-configurable prop.
Therefore, we should remove that key ahead